Refactor IO api for clarity and allow computing chunks directly on index

glopesdev opened the pull request:
There was a confusingly named function called `aeon` remaining from the original first draft of the API. This could be confused with `aeon` the module name, so here we propose deprecating this time conversion function and replace it with two new explicitly named functions `to_datetime` and `to_seconds` instead.

In addition, we extend the `chunk` function to allow extracting chunks directly from a `DatetimeIndex` object.

lochhh reviewed (approved):
Thanks @glopesdev LGTM just a couple of recommendations related to docstrings and we could consider adding simple tests (I can do this in a separate PR as well).

glopesdev commented:
@lochhh Thanks, I fixed all of these and just added an extra code path for converting `pd.Series` objects, which came up while debugging something else. This way most of the relevant round-trip conversion paths should be covered.
